app/main.py

```
@@ -1,44 +1,72 @@
+import asyncio
 import time
-
+from typing import Any, Awaitable
 from iot.devices import HueLightDevice, SmartSpeakerDevice, SmartToiletDevice
 from iot.message import Message, MessageType
 from iot.service import IOTService
 
 
-def main() -> None:
-    # create an IOT service
+async def run_sequence(*functions: Awaitable[Any]) -> None:
+    for function in functions:
+        await function
+
+
+async def run_parallel(*functions: Awaitable[Any]) -> list:
+    return await asyncio.gather(*functions)
```

Minor suggestion: `run_parallel` returns the gathered results. Consider specifying the return type as `list[Any]` for clarity and, optionally, using `asyncio.gather(..., return_exceptions=True)` if you want to collect exceptions instead of failing fast. This is optional but can help robustness when running many tasks in parallel.